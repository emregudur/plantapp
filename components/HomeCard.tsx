import React from 'react'
import { View, Text, Image, ImageBackground, StyleSheet } from 'react-native'

interface Props {
  id: number
  title: string
  subtitle?: string
  image_uri: string
  uri?: string
  order?: number
  isCategory?: boolean
}

export default function HomeCard({ id, image_uri, title, isCategory = false }: Props) {
  return (
    <ImageBackground
      style={isCategory ? styles.categoryContent : styles.content}
      imageStyle={isCategory ? styles.categoryImage : styles.image}
      source={{ uri: image_uri }}
      key={id}
    >
      <Text style={isCategory ? styles.categoryTitle : styles.title}>{title}</Text>
    </ImageBackground>
  )
}

const styles = StyleSheet.create({
  content: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'flex-end',
    width: 240,
    height: 160,
    marginRight: 12,
  },
  categoryContent: {
    display: 'flex',
    width: 158,
    height: 152,
    margin: 12,
  },
  image: {
    borderWidth: 1,
    borderColor: '#FFFFFF1A',
    borderRadius: 12,
  },
  categoryImage: {
    backgroundColor: '#F4F6F6',
    borderWidth: 0.5,
    borderColor: '#29BB892E',
    borderRadius: 12,
  },
  title: {
    color: 'white',
    padding: 20,
    fontFamily: 'Rubik-Medium',
    fontSize: 15,
    fontWeight: '500',
    lineHeight: 20,
    letterSpacing: -0.23,
    textAlign: 'left',
    bottom: 0,
  },
  categoryTitle: {
    width: '100%',
    color: 'black',
    fontFamily: 'Rubik-Medium',
    fontSize: 16,
    fontWeight: '500',
    lineHeight: 21,
    letterSpacing: -0.31,
    textAlign: 'left',
    paddingLeft: 20,
    paddingTop: 20,
  },
})
