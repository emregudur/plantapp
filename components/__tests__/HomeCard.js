import * as React from 'react'
import { render } from '@testing-library/react-native'
import HomeCard from '../HomeCard'

it(`renders correctly`, () => {
  const tree = render(<HomeCard id={'test'} title={'test'} image_uri={'image_uri'} />).toJSON()

  expect(tree).toMatchSnapshot()
})

it(`should render title correctly`, () => {
  const { getByText } = render(<HomeCard id={'test'} title={'test'} image_uri={'image_uri'} />)

  expect(getByText('test').children[0]).toBe('test')
})

it(`should render image correctly`, async () => {
  const uri = 'https://cms-cdn.plantapp.app/6_edbcc6988a/6_edbcc6988a.png'
  const { findByTestId } = render(<HomeCard id={'test'} title={'test'} image_uri={uri} />)

  expect((await findByTestId('image-bg')).props.source.uri).toEqual(uri)
})
