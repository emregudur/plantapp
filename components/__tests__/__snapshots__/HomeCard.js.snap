// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "height": 160,
      "justifyContent": "flex-end",
      "marginRight": 12,
      "width": 240,
    }
  }
>
  <Image
    source={
      {
        "uri": "image_uri",
      }
    }
    style={
      [
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "height": 160,
          "width": 240,
        },
        {
          "borderColor": "#FFFFFF1A",
          "borderRadius": 12,
          "borderWidth": 1,
        },
      ]
    }
    testID="image-bg"
  />
  <Text
    style={
      {
        "bottom": 0,
        "color": "white",
        "fontFamily": "Rubik-Medium",
        "fontSize": 15,
        "fontWeight": "500",
        "letterSpacing": -0.23,
        "lineHeight": 20,
        "padding": 20,
        "textAlign": "left",
      }
    }
  >
    test
  </Text>
</View>
`;
